xapian-core/weight/tfidfweight.cc

```diff
@@ -149,6 +149,10 @@ TfIdfWeight::get_idfn(Xapian::doccount termfreq, char c) const
 	    // All documents are indexed by the term
 	    if (N == termfreq) return 0;
 	    return log((N - termfreq) / termfreq);
+    case 'f':
+        return (1.0/termfreq);
+    case 's':
+        return pow(log(N/termfreq),2.0);
```

Comment:
Your alignment is off here; you may want to adjust your editor configuration, if it can't pick things up automatically from the file (some can scan the buffer when loaded and figure out indentation configuration).